### PR TITLE
Fix request decorator

### PR DESCRIPTION
**Pull request by MattTYXM**

### CODE REVIEW

**Comment by MattTYXM** [comment]
@bethesque this pull request along with [this one](https://github.com/bethesque/pact-mock_service/pull/64) are the PR's that resulted from you requesting that I break them up by bug. I am just looking to find out when these may be considered?

File: lib/pact/consumer_contract/request_decorator.rb [hunk 1]
```diff
@@ -20,25 +20,35 @@ def as_json options = {}
...
       hash[:body]    = body    if request.specified?(:body)
-      hash
+      include_matching_rules? ? with_matching_rules(hash) : hash
     end
 
     private
 
     attr_reader :request
 
     def path
```
Comment (bethesque) [inline lib/pact/consumer_contract/request_decorator.rb]: I don't think these if statements are required. Doing a reification on an object that is not a term will just return that object. I didn't have these lines in my example - did you find a use case that required them?

Comment (bethesque) [inline lib/pact/consumer_contract/request_decorator.rb]: Can you include a test please? I understand that there may not be an existing test, however, a good PR should always leave code in a better state than it started ;)

**Comment by bethesque** [comment]
Thanks for breaking those up, it means questions about one won't block the other.

File: lib/pact/consumer_contract/request_decorator.rb (hunk 1 above)
Comment (MattTYXM) [inline lib/pact/consumer_contract/request_decorator.rb]: Pact::Reification.from_term(request.query) for instance takes the term that defines the matching rule and wipes it out leaving you with the the example value passed with the term. When I get the time again at work I will write a test to exercise this functionality. It is not in my nature to leave functionality untested but this was my first ruby experience and I was a little short on time. I'll notify you when I get back to this.

Comment (bethesque) [inline lib/pact/consumer_contract/request_decorator.rb]: Appreciated! Good work on your first ruby!

**Comment by MattTYXM** [comment]
@bethesque I added specs to support my code changes, I also implemented the code to support query hashes with matchers.

**Comment by bethesque** [comment]
@MattTYXM are you using the ruby gem directly, or via another wrapper?

**Comment by MattTYXM** [comment]
@bethesque I have forkerd your pact-mock-service-npm reporsitory as well. There have been no changes there aside from the necessary changes to make it point at my fork of this repo. If and when you get pact-mock-service-npm updated to point at this most recent version of pact-mock_service I will need to pull that in and I'll wipe out my forks.

**Comment by bethesque** [comment]
Right, so we have a release cascade. I was hoping to do just a beta release, because I'm a bit nervous about this one, as I haven't had time to sit down and generate some pacts with it myself and inspect the differences. I always like to road test critical changes myself before doing a release as I feel I have a responsibility to do safe releases. Do you have any pacts you've generated with your code that you can post for me?

**Comment by MattTYXM** [comment]
@bethesque I don't have anything I can post publicly but I can try to generate something generic and post it.

**Comment by bethesque** [comment]
👍

**Pull request merged**